New feature: Option to use issue import preview API

mdbenito opened the pull request:
This PR:
* adds an option to use the preview API for issue imports. This enables setting the creation and modification date of issues and comments.
* documents it
* adds `settings.github.token_owner`. Required to check ownership of comments against the user creating the new content on GH.

Additionally, it:

* caches a call to getAllBranches()
* Fixes a typing problem in https://github.com/piceaTech/node-gitlab-2-github/blob/7572017dfbcb0a7a2a8917ab6cb99ebe2c173877/src/githubHelper.ts#L858
* Fixes a problem whereby labels are not converted to lowercase before using them in `createIssue()`: https://github.com/piceaTech/node-gitlab-2-github/blob/7572017dfbcb0a7a2a8917ab6cb99ebe2c173877/src/githubHelper.ts#L258

mdbenito commented:
Sorry for merging a couple of fixes into the PR, but I thought them to be minor enough not to warrant separate PRs.

spruce commented on src/githubHelper.ts:
```diff
@@ -280,6 +298,180 @@ export default class GithubHelper {
+    if (props.body && props.body.indexOf('/uploads/') > -1 && !settings.s3) {
+      props.labels.push('has attachment');
+    }
+    await utils.sleep(this.delayInMs);
```
Is it neccessary to sleep here?

spruce commented on src/githubHelper.ts:
```diff
@@ -586,6 +785,11 @@ export default class GithubHelper {
 
     // Failing all else, create an issue with a descriptive title
+
+    let userIsCreator = pullRequest.author && 
+      ((settings.usermap && settings.usermap[pullRequest.author.username] === settings.github.token_owner) ||
+       (pullRequest.author.username === settings.github.token_owner));
```
You check that multiple times. Doesn't it make sense to extract it into its own function?

mdbenito replied:
yes, it is checked twice and I think I even factored it out into a function but it wasn't exactly the same functionality, then I simplified it and now it is indeed the same 😅 I'll fix that too.

spruce reviewed (commented):
Looks nice. Thanks for work :)

mdbenito commented:
I addressed your points and fixed the label case conversion to only happen if `settings.conversion.useLowerCaseLabels` is true. I am unsure about the return when `settings.debug` is true, though, but it is exactly the same behaviour as in the previous `createIssue`, so that should be ok.
